### DISCUSSION TITLE
Animation file (.h3d) file format description

### DISCUSSION THREAD

**Discussion by nawijn** [OpenRadioss/OpenRadioss#552] [Q&A]
Dear All,

As an exercise for my programming skills in C++ and (hopefully) my contribution back to the OpenRadioss community, I started to work on another conversion tool from OpenRadioss (.h3d) to VTK.  The biggest difference with respect to the `anim_to_vtk` tool currently available, is that I will not write the ASCII output for VTK directly. Instead I will use the VTK API to construct an unstructured grid with all the relevant point and cell data attached and subsequently use the VTK facilities for exporting it to any of the supported formats. 

I currently try to reverse engineer the animation file format from the source code of `anim_to_vtk`. However, it would be very helpful if the there is a documented version of the file format. So the question is: is this documented version of the animation file format available for public use?

Kind regards,

Marco

P.S. If you have a feature request for the new tool, let me know.

**Comment by MarianBulla** [comment]
Dear Marco,
this is really great ! 
One point in the current conversion from animation files to VTK is the fact, that it creates one big model, without the possibility of separately selecting parts (blocks) within ParaView. Usually, the OpenRadioss models consist (and are structured) in subsystems and further on in parts.
The information about the individual part ID is already there. The source code is quite small. Unfortunately, we don't have any specific documentation on it. Sorry for that.
In case of any question, don't hesitate and reach out to me.
Again, thank you very much for your effort on that.
Thanks and best regards,
Marian

**Reply by nawijn** [reply]
Hello Marian,

Thanks for the response! 

In theory, the system/subsystem/part hierarchy should map fairly natural to the  (relatively) new VTK [data assembly](https://vtk.org/doc/nightly/html/md__builds_gitlab-kitware-sciviz-ci_Documentation_Doxygen_DataAssembly.html) facilities. I guess I will find out shortly how theory maps to practice. 

Regarding the layout of the file, I will try and keep notes regarding the reverse engineering of the layout of the animation file so other people can check for correctness and completeness. I will keep you informed on the progress.

I already have a Github placeholder repository (https://github.com/nawijn/operato). No real things to show there yet, but if you want to track progress, you can. 

Marco

**Reply by MarianBulla** [reply]
Hello Marco,
this is really great and much appreciated.
Yes, I definitely will jump on your GitHub repository and have a look at it. Thanks for that info.
Thanks and best regards,
Marian

**Comment by Serince** [comment]
Hello Marco,

I have already started to write some python codes to read anime files. I was planning to write a d3plot converter. Because ascii animation results slowly playing in paraview. Ls-prepost (reads d3plot files) which is the lsdyna pre and post processor is freely available and huge animation files playing nicely. My script is not finished, again I will upload to github.

All the best,
Serdar

**Reply by MarianBulla** [reply]
Hello Serdar,
yes, sounds also really interesting.
Thank you for your effort on this.

Thanks and best regards,
Marian

**Reply by Serince** [reply]
Hello All,

You can find unfinished version of anime file reader script  in Python at the following link
https://github.com/Serince/TVss/

All the best,
Serdar

**Reply by ehan08** [reply]
Hello Serdar,

Thanks for the nice project. I would like to join. Is the format of the animation file available online?

Thanks.

Ed

**Reply by Serince** [reply]
Hello Ed,

I'd be very happy if you joined me. I couldn't find any documentation for anime files. However, it is not difficult to understand the structure of anime files through the anime_to_vtk converter under src/tools. 
I have wrote the reading anime file of the anime_to_vtk file in the python. Now there are 2 things to do next. First is to test the file in different result files. Second, reading T01 file with python according to th_to_csv file. After these processes are completed, I will start to code d3plot file writer. 

Best regards, 
Serdar

**Comment by prince17080** [comment]
Hi @MarianBulla, could you please suggest how we can convert the .h3d file format to either .vtk or .anim file format? It would be really helpful.

**Reply by MarianBulla** [reply]
Hi @prince17080, 
I don't know any way, to convert the .h3d file to any other. 
But you can remove the /H3D/ - requests from your engine file and add /ANIM/ - requests, instread.

For more information, please have a look here: 
https://help.altair.com/hwsolvers/rad/topics/solvers/rad/engine_input_r.htm

Hope it helps you ?

Thanks and best regards,
Marian
